activerecord/test/cases/strict_loading_test.rb

```diff
@@ -672,6 +673,18 @@ def test_strict_loading_violation_logs_on_polymorphic_relation
     ActiveRecord.action_on_strict_loading_violation = old_value
   end
 
+  def test_strict_loading_in_n_plus_one_only_mode_does_not_eager_load_child_associations
```

Comment:
Could you move this test under `test_strict_loading_n_plus_one_only_mode`, so all `:n_plus_one_only` tests are grouped together?
I've also split that test into two separate tests. You probably need to rebase on main.

Reply:
Of course. Moved in d32ce805bf7ad7eeeb910afbe632ba66d48707d0. I also tweaked the test name slightly to better align with the others.